### PR TITLE
Let Hosts default to redirecting on port 8080

### PR DESCRIPTION
**Pull request by kflynn**
This restores the de facto behavior of 1.0 and 1.1 -- a `Host` resource using TLS would default to supporting redirection from 8080 to 8443. You can now disable that behavior by setting `insecure.addtionalPort` to -1, and the KAT tests do a better job of testing for it.

This is probably best reviewed commit-by-commit. I tested with KAT under OSS and Edge Stack, and by running the migration scenario by hand.

### CODE REVIEW

**Comment by LukeShu** [comment]
Oh, also: This needs a doc change.

File: python/tests/t_hosts.py [hunk 1]
```diff
@@ -136,24 +144,32 @@ def scheme(self) -> str:
...
+                    minTLSv="v1.0",  maxTLSv="v1.0",
+                    error=["tls: server selected unsupported protocol version 303",
+                           "tls: no supported versions satisfy MinVersion and MaxVersion",
+                           "tls: protocol version not supported"])
 
-class HostClearText(AmbassadorTest):
+        if EDGE_STACK:
+            yield Query(self.url("target/", scheme="http"), expected=404)
+        else:
+            yield Query(self.url("target/", scheme="http"), error=[ "EOF", "connection refused" ])
```
[the remark this answers is not in the document]
Comment (kflynn) [reply python/tests/t_hosts.py]: Yup.

File: api/getambassador.io/v2/Host.proto [hunk 2]
```diff
@@ -126,7 +126,7 @@ message InsecureRequestPolicy {
   InsecureRequestAction action = 1;
 
   // Is there an additional insecure port we should listen on?
-  uint32 additionalPort = 2;
+  int32 additionalPort = 2;
```
[the remark this answers is not in the document]
Comment (kflynn) [reply api/getambassador.io/v2/Host.proto]: !*@&#!*@&#*!@#& 

Yup, you're right. OK, force-push coming soon.

File: python/ambassador/envoy/v2/v2listener.py [hunk 3]
```diff
@@ -1077,21 +1077,19 @@ def generate(cls, config: 'V2Config') -> None:
...
+                            # We have to force the correct route entry, too, just in case. (Note that right now,
+                            # the user can't create a Mapping that forces redirection. When they can do this
+                            # per-Mapping, well, really, we can't force them to not redirect if they explicitly
+                            # ask for it, and that'll be OK.)
+
                             if secure:
-                                logger.debug(
-                                    f"V2Listeners: {listener.name} {vhostname} secure: force Reject for ACME challenge")
-                                action = "Reject"
+                                route = secure_route
```
Comment (LukeShu) [inline python/ambassador/envoy/v2/v2listener.py]: Is it even possible for `route` to not already be `secure_route` if we get here? (and for `action` to not already be `"Route"` a few lines up?

Am I correct in thinking that
```python
if route["match"].get("prefix", None) == "/.well-known/acme-challenge/":
```
could become
```
if route["match"].get("prefix", None) == "/.well-known/acme-challenge/" and not secure:
```
and then drop the `if secure` bit?  And that it wouldn't change behavior at all?

Comment (LukeShu) [reply python/ambassador/envoy/v2/v2listener.py]: Flynn says he is 99.5% certain that what I said is accurate.

**Pull request merged**